ssvm-core/src/main/java/dev/xdark/ssvm/filesystem/natives/GenericFileSystemNatives.java

```diff
@@ -249,6 +222,77 @@ public void init(VirtualMachine vm, FileManager fileManager) {
 			}
 			return Result.ABORT;
 		});
+
+		InstanceClass raf = (InstanceClass) vm.findBootstrapClass("java/io/RandomAccessFile");
+		if (raf != null) {
+			JavaField pathField = raf.getField("path", "Ljava/lang/String;");
+			vmi.setInvoker(raf, "initIDs", "()V", MethodInvoker.noop());
+			vmi.setInvoker(raf, "length", "()J", ctx -> {
+				Locals locals = ctx.getLocals();
+				InstanceValue _this = locals.loadReference(0);
+				String pathLiteral = ops.toString(vm.getMemoryManager().readReference(_this, pathField.getOffset()));
+				try {
+					long size = fileManager.getAttributes(pathLiteral, BasicFileAttributes.class).size();
+					ctx.setResult(size);
+				} catch (FileNotFoundException ex) {
+					ops.throwException(vm.getSymbols().java_io_FileNotFoundException(), ex.getMessage());
+				} catch (IOException ex) {
+					ops.throwException(vm.getSymbols().java_io_IOException(), ex.getMessage());
+				}
+				return Result.ABORT;
+			});
+			vmi.setInvoker(raf, "readBytes", "([BII)I", ctx -> readBytes(ctx, vm, fileManager));
+			vmi.setInvoker(raf, "seek0", "(J)V", ctx -> {
+				Locals locals = ctx.getLocals();
+				InstanceValue _this = locals.loadReference(0);
+				long handle = getFileStreamHandle(vm, _this);
+				InputStream in = fileManager.getFdIn(handle);
+				if (in != null) {
+					try {
+						long pos = locals.loadLong(1);
+						in.reset();
+						int skipped = 0;
+						while (skipped < pos) {
+							skipped += in.skip(pos - skipped);
+						}
+					} catch (IOException ex) {
+						ops.throwException(vm.getSymbols().java_io_IOException(), ex.getMessage());
+					}
+				}
+				return Result.ABORT;
+			});
+			vmi.setInvoker(raf, "open0", "(Ljava/lang/String;I)V", ctx -> {
+				Locals locals = ctx.getLocals();
+				InstanceValue _this = locals.loadReference(0);
+				String path = ops.readUtf8(locals.loadReference(1));
+				int mode = locals.loadInt(2);
+				try {
+					int fmMode = FileManager.READ;
+					if (mode == 2) fmMode = FileManager.ACCESS_READ | FileManager.ACCESS_WRITE;
+
+					// TODO: Temporary hack
+					//  - The context check here is so that we can determine if this is being done from a ZipFile constructor
+					//  - We want to open this as a zip file in our file manager so other Jar/ZipFileNatives hooks function correctly
+					//  - However, this is obviously a huge hack. I'm not sure how we should best approach this.
+					long handle;
+					if (ctx.getVM().getThreadManager().currentOsThread().getBacktrace().at(9).getMethod().toString().contains("java/util/zip/ZipFile.<init>")) {
+						// Zip file modes are different
+						fmMode = ZipFile.OPEN_READ;
+						handle = fileManager.openZipFile(path, fmMode);
+					} else {
+						handle = fileManager.open(path, fmMode);
+					}
```

Comment:
This whole block here needs major work done on it.... I'm not sure how to best approach this issue though.